src/lib.rs

```diff
@@ -140,14 +157,18 @@ impl Lottery {
             let win_player_index = *self.players.keys().nth(index).expect("Player not found");
             let player = self.players[&win_player_index];
 
-            if self.token_address.is_some() {
+            if let Some(token_address) = self.token_address {
                 debug!("Transfer tokens to the winner");
-                self.transfer_tokens(&exec::program_id(), &player.player_id, self.lottery_balance)
-                    .await;
-
-                self.lottery_balance = 0;
+                self.transfer_tokens(
+                    &token_address,
+                    &exec::program_id(),
+                    &player.player_id,
+                    self.lottery_state.prize_fund,
+                )
+                .await;
             } else {
-                msg::send_bytes(player.player_id, b"Winner", exec::value_available()).unwrap();
+                msg::send_bytes(player.player_id, b"Winner", self.lottery_state.prize_fund)
+                    .unwrap();
```

Comment:
A good practice is to use `expect("error message")` instead of `unwrap()` only.

Reply:
Fixed